### PR TITLE
Added new PR with optimized sign In and sign up functionality

### PR DESCRIPTION
**Pull request by AdityaPainuli**
Hey this one is the new PR I fixed the bugs and now you can switch between sign-in and sign-up very easily . let me know if you want any change for the google login button I am still working on it.

### CODE REVIEW

**Comment by ykdojo** [comment]
@AdityaPainuli can you double-check and make sure to address all the points I raised in the previous PR? #64

Also, could you please use proper punctuation? I highly recommend Grammarly to check everything you write before you send it here.

**Comment by rohitdasu** [comment]
why we need mui? we already have tailwindCSS,
not a good idea to have two things at once. @ykdojo what's your opinion?

**Comment by ykdojo** [comment]
@rohitdasu right, we should probably stick with one unless they're designed to work together.

If we were to pick one, Tailwind seems like a good choice. Open to suggestions, though

**Comment by rohitdasu** [comment]
@ykdojo tailwindCSS is enough to work with the project. If we need any components then TailwindCSS provides components as well and we can create our custom components using headless ui which is provided by TailwindCSS. So we don't need MUI or other libraries. It will make our app unnecessarily heavy.

**Comment by AdityaPainuli** [comment]
Ok, I will fix the issues so do we need to create different routes for the sign-in and sign-up pages? Also, I think it will be great if the page doesn't get loaded when the user switches between those two what do you think? let me know I will do things accordingly

**Comment by AdityaPainuli** [comment]
> @ykdojo tailwindCSS is enough to work with the project. If we need any components then TailwindCSS provides components as well and we can create our custom components using headless ui which is provided by TailwindCSS. So we don't need MUI or other libraries. It will make our app unnecessarily heavy.

What should be the best for icons cause in hero icons which is the tailwind library I searched for google icons and couldn't able to find them. Sorry , If I am asking silly question

**Comment by iShibi** [comment]
@AdityaPainuli Your `CHANGELOG.md` contains snippets of git conflicts. I suggest you to delete the file since it's already been removed from `main`. A better option will be to delete this fork and re-fork the repo, then create a new branch (you are working in the `main` of you fork currently, not recommended) and add these changed back in a new PR.

**Comment by rohitdasu** [comment]
> > @ykdojo tailwindCSS is enough to work with the project. If we need any components then TailwindCSS provides components as well and we can create our custom components using headless ui which is provided by TailwindCSS. So we don't need MUI or other libraries. It will make our app unnecessarily heavy.
> 
> What should be the best for icons cause in hero icons which is the tailwind library I searched for google icons and couldn't able to find them. Sorry , If I am asking silly question

Don't hesitate to question, it's a good habit. Regarding the icons package if we don't get the desired icon from heroicons then we can get the icons from FontAwesome website. 
NOTE: we should not install the library, instead download the icon(SVG) from FontAwesome website and put it in our public folder. 
any suggestions ? 
CC: @iShibi @ykdojo

**Comment by AdityaPainuli** [comment]
Thanks, I will keep that in mind

**Comment by rohitdasu** [comment]
Also this image is breaking in welcome screen

Screenshot 2022-08-14 at 10 04 35 AM

**Comment by AdityaPainuli** [comment]
> Ok, I will fix the issues so do we need to create different routes for the sign-in and sign-up pages? Also, I think it will be great if the page doesn't get loaded when the user switches between those two what do you think? let me know I will do things accordingly

What do you guys think about this? should I create a different page or the same page?

**Comment by AdityaPainuli** [comment]
I think I put a link in it that's why this is breaking up should I download the image or is there any other way ??
image

**Comment by rohitdasu** [comment]
> I think I put a link in it that's why this is breaking up should I download the image or is there any other way ?? image

if you have permission to use that image then download and put it inside public folder

**Comment by AdityaPainuli** [comment]
Ok, this Image is from unsplash.com and is available for free to use. let me correct all the issues .

**Comment by AdityaPainuli** [comment]
Fixed the broken image link.

**Comment by ykdojo** [comment]
For routing, first of all, a smaller PR is generally better. So can you keep the routes the same as before in this PR? In this PR, you can focus on the visual elements.

You're welcome to raise an issue, a discussion, or a PR for the routes if you think we should change them later.

Also, as I said earlier, could you please use correct punctuation in your comments for better readability?

For example, here:

> Ok, this Image is from unsplash.com and is available for free to use. let me correct all the issues .

There should not be a space before the second period, and the beginning of each sentence should be capitalized.

I recommend Grammarly for this.

**Comment by AdityaPainuli** [comment]
Ok sorry for the inconvenience. I will try to keep these things in my mind.

**Comment by AdityaPainuli** [comment]
I created a different page for sign up and sign in as you said. Let me know what you think about it now.

**Comment by rohitdasu** [comment]
i can't access the signin and signup page - @AdityaPainuli

**Comment by AdityaPainuli** [comment]
Yes, it's not deployed yet . You can see the personal deployed version If you want and let me know if it is good.
[https://defaang-eight.vercel.app/](url)

File: package.json [hunk 1]
```diff
@@ -11,7 +11,12 @@
 		"prepare": "husky install"
 	},
 	"dependencies": {
+		"@emotion/react": "^11.10.0",
+		"@emotion/styled": "^11.10.0",
+		"@mui/icons-material": "^5.8.4",
+		"@mui/material": "^5.10.0",
```
Comment (rohitdasu) [inline package.json:17]: please remove MUI

File: pages/signin.tsx [hunk 2]
```diff
@@ -1,66 +1,103 @@
+import GoogleIcon from '@mui/icons-material/Google';
```
Comment (rohitdasu) [inline pages/signin.tsx:1]: don't use mui

**Review by rohitdasu — CHANGES_REQUESTED** [review]
please work on the changes requested

File: package.json (hunk 1 above)
Comment (AdityaPainuli) [reply package.json:17]: Ok, I will work on that.

**Comment by AdityaPainuli** [comment]
I removed Mui completely from the project.

**Comment by rohitdasu** [comment]
please use semantic commits - @AdityaPainuli 
for more info read project's contribution guide

**Comment by AdityaPainuli** [comment]
Ok, I will keep that in mind for the next time.

File: pages/signin.tsx [hunk 3]
```diff
@@ -1,66 +1,107 @@
 import { useState } from 'react';
-import type { NextPage } from 'next';
+import Image from 'next/image';
 import Head from 'next/head';
+import { NextPage } from 'next';
```
Comment (iShibi) [inline pages/signin.tsx:4]:
```suggestion
import { type NextPage } from 'next';
```

File: pages/signup.tsx [hunk 4]
```diff
@@ -1,66 +1,132 @@
 import { useState } from 'react';
-import type { NextPage } from 'next';
+import Image from 'next/image';
 import Head from 'next/head';
+import { NextPage } from 'next';
```
Comment (iShibi) [inline pages/signup.tsx:4]:
```suggestion
import { type NextPage } from 'next';
```

**Review by iShibi — CHANGES_REQUESTED** [review]
Import `NextPage` as a type so it's not included in the bundle

**Comment by AdityaPainuli** [comment]
@ykdojo can you review the code and let me know what are your thoughts?

**Comment by ykdojo** [comment]
@AdityaPainuli can you remove the Google button since it's not working yet?

For the login / signup, did you test the functionality?

**Comment by ykdojo** [comment]
Also, in the UI text, I spotted some grammatical mistakes. Can you check them?

**Comment by AdityaPainuli** [comment]
Sure I will work on that .

**Comment by AdityaPainuli** [comment]
I checked for the sign-in and sign-up functionality it's working fine with the database. also applied the changes you mentioned.

File: pages/signup.tsx [hunk 5]
```diff
@@ -1,66 +1,122 @@
...
+						</form>
+						<button
+							type='submit'
+							onClick={handleSignup}
+							className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+						>
+							Login
+						</button>
+						<p className='py-2 font-semibold'>
+							Already has an account ?{' '}
```
Comment (ykdojo) [inline pages/signup.tsx:97]: has -> have

remove the space before the question mark

Comment (ykdojo) [reply pages/signup.tsx:97]: > {' '}

What's this for?

File: pages/signin.tsx [hunk 6]
```diff
@@ -1,66 +1,97 @@
...
+							onSubmit={e => {
+								e.preventDefault();
+								handleSignIn();
+							}}
+						>
+							<div className='flex w-[90%] flex-col space-y-2'>
+								<p>Email</p>
+								<input
+									type='text'
+									placeholder='dojo@gmaill.com'
```
Comment (ykdojo) [inline pages/signin.tsx:45]: for a placeholder, it's a good practice to use the example.com domain.

File: pages/signup.tsx [hunk 7]
```diff
@@ -1,66 +1,122 @@
...
+								e.preventDefault();
+								handleSignup();
+							}}
+							className='mt-4 flex flex-col space-y-4'
+						>
+							<div className='flex w-[90%] flex-col space-y-2'>
+								<p>Email</p>
+								<input
+									type='text'
+									placeholder='dojo@gmaill.com'
```
Comment (ykdojo) [inline pages/signup.tsx:48]: same here

File: pages/signin.tsx [hunk 8]
```diff
@@ -1,66 +1,97 @@
...
+						</form>
+						<button
+							type='submit'
+							className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+							onClick={handleSignIn}
+						>
+							Login
+						</button>
+						<p className='py-2 font-semibold'>
+							Not registered yet ?{' '}
```
Comment (ykdojo) [inline pages/signin.tsx:72]: remove the space

> {' '}

what's this for?

File: pages/signin.tsx [hunk 9]
```diff
@@ -1,66 +1,97 @@
...
+									onChange={e => setPassword(e.target.value)}
+								/>
+							</div>
+						</form>
+						<button
+							type='submit'
+							className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+							onClick={handleSignIn}
+						>
+							Login
```
Comment (ykdojo) [inline pages/signin.tsx:69]: can you use 2 spaces for tabs?

like this: https://github.com/ykdojo/defaang/commit/30baa13463639ca151526e205fadecd7653d6e90

**Review by ykdojo — CHANGES_REQUESTED** [review]
commented on some of the more obvious things.

we might need to do this a few more rounds - let's see

**Comment by ykdojo** [comment]
also, it looks like there are merge conflicts now

**Review by Dun-sin — COMMENTED** [review]
It's not good to commit package.json or package-lock.json unless you are installing a new package that is being used for the project

**Comment by AdityaPainuli** [comment]
Hey, I corrected the lines you mentioned but it's saying that this branch has conflicts . Can you help me how to resolve them

**Comment by AdityaPainuli** [comment]
@iShibi can you review the code and let me know what is the error?

**Comment by iShibi** [comment]
> @iShibi can you review the code and let me know what is the error?

Your `package-lock.json` seems to be out of sync. Delete it and run `npm i`. For the prettier error wait for:
- #80

**Comment by AdityaPainuli** [comment]
Fixed the correction and conflicts as @ykdojo mentioned. Please review the code and let me know what you think.

File: pages/signup.tsx [hunk 10]
```diff
@@ -1,67 +1,125 @@
...
+						</form>
+						<button
+							type='submit'
+							onClick={handleSignup}
+							className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+						>
+							Login
+						</button>
+						<p className='py-2 font-semibold'>
+							Already have an account ?
```
Comment (ykdojo) [inline pages/signup.tsx:98]: remove the space

File: pages/signup.tsx [hunk 11]
```diff
@@ -1,67 +1,125 @@
...
+							<span
+								className='cursor-pointer font-bold text-blue-600 hover:underline'
+								onClick={() => router.push('/signin')}
+							>
+								Sign in
```
Comment (ykdojo) [inline pages/signup.tsx:103]: add a space before "Sign in" and make sure it shows up on the UI

File: pages/signin.tsx [hunk 12]
```diff
@@ -1,67 +1,99 @@
...
+						</button>
+						<p className='py-2 font-semibold'>
+							Not registered yet ?
```
Comment (ykdojo) [inline pages/signin.tsx:73]: remove the space

File: pages/signin.tsx [hunk 13]
```diff
@@ -1,67 +1,99 @@
...
+							<span
+								className='cursor-pointer font-bold text-blue-600 hover:underline'
+								onClick={() => router.push('/signup')}
+							>
+								Register here
```
Comment (ykdojo) [inline pages/signin.tsx:78]: add a space before "Register here" and make sure it shows up on the UI

also, we should say "Sign up" instead of "Register"

**Review by ykdojo — CHANGES_REQUESTED** [review]
Added a few comments.

Another thing - in the login view, it looks squished for some reason (for example, there should be more padding between the Password label and the email box):

image

File: pages/signin.tsx (hunk 13 above)
Comment (AdityaPainuli) [reply pages/signin.tsx:78]: Actually {" "} this was giving up the space which we removed earlier. 😅

**Comment by AdityaPainuli** [comment]
Applied all the corrections as you mentioned above @ykdojo

File: pages/signup.tsx [hunk 14]
```diff
@@ -1,66 +1,123 @@
...
+            </form>
+            <button
+              type='submit'
+              onClick={handleSignup}
+              className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+            >
+              Login
+            </button>
+            <p className=' font-semibold'>
+              Already have an account ?
```
Comment (ykdojo) [inline pages/signup.tsx:97]: there's still a space before the question mark. I believe this is the third time I said this

File: pages/signin.tsx [hunk 15]
```diff
@@ -1,66 +1,98 @@
...
+            </form>
+            <button
+              type='submit'
+              className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+              onClick={handleSignIn}
+            >
+              Login
+            </button>
+            <p className=' font-semibold'>
+              Not registered yet ?
```
Comment (ykdojo) [inline pages/signin.tsx:72]: same thing here

**Review by ykdojo — CHANGES_REQUESTED** [review]

File: pages/signup.tsx (hunk 14 above)
Comment (AdityaPainuli) [reply pages/signup.tsx:97]: Really sorry I assumed you are mentioning padding on the y-axis I will correct it now.

**Comment by AdityaPainuli** [comment]
Is it okay now? Sorry I didn't understand it at the first.

File: pages/signin.tsx (hunk 8 above)
Comment (rohitdasu) [reply pages/signin.tsx]: @ykdojo i guess it is added by prettier. it's for blank space.

File: next.config.js [hunk 16]
```diff
@@ -1,4 +1,9 @@
 /** @type {import('next').NextConfig} */
 module.exports = {
-  reactStrictMode: true,
+
+	reactStrictMode: true,
+	images: { domains: ['images.unsplash.com', 'cdn-icons-png.flaticon.com'] },
```
Comment (ykdojo) [inline next.config.js:5]: instead of linking unsplash directly, can you download the image and put it in your fork?

probably the same thing with the icon

**Review by ykdojo — CHANGES_REQUESTED** [review]

File: pages/signin.tsx [hunk 17]
```diff
@@ -1,66 +1,98 @@
...
-            value={email}
-            onChange={e => setEmail(e.target.value)}
-          />
+      <main className='flex h-screen w-screen'>
+        {/* Left section - main login  */}
+
+        <div className='flex h-screen  w-[100%] flex-col items-center justify-center bg-gray-100 lg:w-[50%]'>
+          {/* Now you can decide different login ways - primarily I am giving google and twiiter for more add it yourself. */}
+          <div className='w-full  bg-white p-4 md:shadow-lg lg:h-[500px] lg:w-[400px]'>
+            <h1 className='p-2 text-3xl font-semibold'>Login ⚡</h1>
```
Comment (ykdojo) [inline pages/signin.tsx:32]: let's use *Sign in* instead of *Login*

Comment (AdityaPainuli) [reply pages/signin.tsx:32]: Should I also do the same to create an account

Comment (ykdojo) [reply pages/signin.tsx:32]: Create an account is fine, I think?

**Comment by AdityaPainuli** [comment]
As we are not using google icon right now I removed it from the project also the image link instead downloaded an image in the public folder.  Is it ok now @ykdojo

**Comment by ykdojo** [comment]
Should be okay now assuming the signing up and signing in still work.

I would like to wait for one or two of the other maintainers to check this first though.

File: pages/signup.tsx [hunk 18]
```diff
@@ -1,66 +1,117 @@
...
+                  onChange={e => setConfirmPass(e.target.value)}
+                />
+              </div>
+            </form>
+            <button
+              type='submit'
+              onClick={handleSignup}
+              className='my-4 flex  w-[90%] cursor-pointer items-center  justify-center rounded-md bg-blue-600 py-2 font-semibold text-white hover:bg-blue-500'
+            >
+              Login
```
Comment (rohitdasu) [inline pages/signup.tsx:93]: change login to "signup "

File: pages/signup.tsx [hunk 19]
```diff
@@ -1,66 +1,117 @@
...
-  const [password, setPassword] = useState<string>('');
-
-  const handleSignUp = async () => {
+  const router = useRouter();
+  const [email, setEmail] = useState('');
+  const [password, setPassowrd] = useState('');
+  const [confirmPass, setConfirmPass] = useState('');
+  const [username, setUsername] = useState('');
+  const handleSignup = async () => {
+    console.log('ok');
```
Comment (rohitdasu) [inline pages/signup.tsx:14]: remove console.log

File: pages/signup.tsx [hunk 20]
```diff
@@ -1,66 +1,117 @@
...
-            type='password'
-            placeholder='Password'
-            value={password}
-            onChange={e => setPassword(e.target.value)}
-          />
+        {/* Right Section - Image and just for UI. */}
+
+        <div className='relative hidden w-[50%] lg:inline-flex'>
+          <div className='absolute top-0 left-0 h-[100%] w-[100%] bg-black opacity-[0.5]'></div>
+          <Image src={require('../public/sidebar.jpg')} alt='right-side-image' layout='fill' className='object-cover' />
```
Comment (rohitdasu) [inline pages/signup.tsx:111]:
```suggestion
          <Image src="/sidebar.jpg" alt='right-side-image' layout='fill' className='object-cover' />
```

**Review by rohitdasu — CHANGES_REQUESTED** [review]

**Review by rohitdasu — APPROVED** [review]
I've checked the auth part, it was working! but UI is what i'm not fully satisfied but it can be improved later when we have proper designs. But anyways good job @AdityaPainuli 
I would suggest other maintainers also to review it. Thanks

**Comment by subhoghoshX** [comment]
Is there any reason you used next router instead of the `Link` component?

**Comment by AdityaPainuli** [comment]
No, I don't think there is any specific reason for it I just want to use it. Is there any benefit in using a link instead of a router?

**Comment by rohitdasu** [comment]
> No, I don't think there is any specific reason for it I just want to use it. Is there any benefit in using a link instead of a router?

if there is no conditional routing then we can simply use Link component.

**Comment by AdityaPainuli** [comment]
Then should I change it to a link component?

**Comment by rohitdasu** [comment]
> Then should I change it to a link component?

yeah it should be changed if there is a simple route exist we should use Link only

**Comment by rohitdasu** [comment]
also please resolve branch conflicts

**Comment by subhoghoshX** [comment]
I think removing it will simplify the code. Unless we're planning to redirect the user to the sign-in page after sign-up.

After signing up, if we immediately give the user access to the dashboard, then it's not necessary imo. But if we add another step in between, i.e. redirect the user to the sign-in page, then it might be useful.

**Comment by rohitdasu** [comment]
> I think removing it will simplify the code. Unless we're planning to redirect the user to the sign-in page after sign-up.
> 
> After signing up, if we immediately give the user access to the dashboard, then it's not necessary imo. But if we add another step in between, i.e. redirect the user to the sign-in page, then it might be useful.

yeah it depends to the use case

**Comment by subhoghoshX** [comment]
Let's merge it for now. We can refactor later. But it seems there are merge conflicts. @AdityaPainuli mind taking a look at it?

**Pull request merged**